### PR TITLE
parser: add warning when the hint in the wrong place

### PR DESCRIPTION
**Pull request by Reminiscent**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/42149

Problem Summary:
add warning when the hint in the wrong place

### What is changed and how it works?
add warning when the hint in the wrong place

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by winoros — APPROVED** [review]

File: planner/core/casetest/integration_test.go [hunk 1]
```diff
@@ -65,6 +65,22 @@ func TestPushLimitDownIndexLookUpReader(t *testing.T) {
...
+	tk.MustExec("set tidb_cost_model_version=2")
+	tk.MustExec("use test")
+	tk.MustExec("drop table if exists t")
+	tk.MustExec("create table t(a int, b int)")
+
+	tk.MustExec("select /*+ stream_agg() */ count(*) from t where a > 1;")
+	tk.MustQuery("show warnings").Check(testkit.Rows())
+
+	tk.MustExec("select count(*) /*+ stream_agg() */ from t where a > 1;")
+	tk.MustQuery("show warnings").Check(testkit.Rows("Warning 1064 You have an error in your SQL syntax; check the manual that corresponds to your TiDB version for the right syntax to use [parser:8066]Optimizer hint can only be followed by certain keywords like SELECT, INSERT, etc."))
```
Comment (fzzf678) [inline planner/core/casetest/integration_test.go:81]: Can this warning take effect in `insert into  ... select /*+ hint()*/ ...` and subquery

**Review by fzzf678 — APPROVED** [review]

**Review by qw4990 — APPROVED** [review]

**Pull request merged**